linker/Linker/Annotations.cs

```diff
@@ -87,7 +87,7 @@ public AssemblyAction GetAction (AssemblyDefinition assembly)
 			if (assembly_actions.TryGetValue (assembly, out action))
 				return action;
 
-			throw new NotSupportedException ();
+			throw new InvalidOperationException($"No action for the assembly {assembly.Name.Name} defined");
```

Comment:
Use ToString to include full name

```suggestion
			throw new InvalidOperationException($"No action for the assembly {assembly.Name} defined");
```